File: src/main/scala/TSVToParquet.scala
import org.apache.spark.sql.SparkSession

object TSVToParquet {
       def main(args: Array[String]) {
           val spark = SparkSession.builder.appName("TSV to Parquet").getOrCreate()
           val df = spark.read.format("csv").option("delimiter", "\t").option("header", "true").load(args(0))
           df.write.parquet(args(1))
       }
}

File: src/main/java/TSVToParquet.java
import org.apache.spark.sql.*;

public class TSVToParquet {
    public static void main(String[] args) {
        SparkSession spark = SparkSession.builder().appName("TSVToParquet").getOrCreate();
        Dataset<Row> df = spark.read().format("csv").option("delimiter", "\t").option("header", "true").load(args[0]);
        df.write().parquet(args[1]);
    }
}
